mailbox/api/src/main/java/org/apache/james/mailbox/model/SimpleMailboxAnnotation.java

```diff
@@ -0,0 +1,328 @@
+package org.apache.james.mailbox.model;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.regex.Pattern;
+
+import org.apache.james.mailbox.exception.UnsupportedAnnotationException;
+import org.apache.james.mailbox.model.MailboxACL.MailboxACLEntryKey;
+
+import com.google.common.base.Preconditions;
+
+public class SimpleMailboxAnnotation implements MailboxAnnotation {
+
+    private static final Pattern NAME_ANNOTATION_PATTERN = Pattern.compile("[^\\x1a-\\x7f]");
+
+    public static final MailboxAnnotation EMPTY;
+
+    private final Map<MailboxAnnotationEntryKey, MailboxAnnotationEntryValue> entries;
+
+    static {
+        EMPTY = new SimpleMailboxAnnotation();
+    }
+
+    /**
+     * Creates a new instance of SimpleMailboxAnnotation containing no entries.
+     * 
+     */
+    public SimpleMailboxAnnotation() {
+        this.entries = Collections.emptyMap();
+    }
+
+    /**
+     * Creates a new instance of SimpleMailboxAnnotation from the given array of
+     * entries.
+     * 
+     * @param entries
+     */
+    public SimpleMailboxAnnotation(Map.Entry<MailboxAnnotationEntryKey, MailboxAnnotationEntryValue>[] entries) {
+        if (entries != null) {
+            Map<MailboxAnnotationEntryKey, MailboxAnnotationEntryValue> m = 
+                    new HashMap<MailboxAnnotationEntryKey, MailboxAnnotationEntryValue>(entries.length + entries.length / 2 + 1);
+            for (Entry<MailboxAnnotationEntryKey, MailboxAnnotationEntryValue> en : entries) {
+                m.put(en.getKey(), en.getValue());
+            }
+            this.entries = Collections.unmodifiableMap(m);
+        } else {
+            this.entries = Collections.emptyMap();
+        }
+    }
+
+    /**
+     * Creates a new instance of SimpleMailboxAnnotation from the given {@link Map} of
+     * entries.
+     * 
+     * @param entries
+     */
+    public SimpleMailboxAnnotation(Map<MailboxAnnotationEntryKey, MailboxAnnotationEntryValue> entries) {
+        if (entries != null && entries.size() > 0) {
```

Comment:
there's a lot of duplication here. You usually should have a single constructor that initialize fields and all others should call that one. If constructor syntax is too cumbersome, you have to switch to static method factory pattern.